Source/Core/Core/IOS/Network/KD/NetKDRequest.h

```diff
@@ -52,12 +53,66 @@ class NetKDRequestDevice : public Device
     return std::nullopt;
   }
 
-  NWC24::NWC24Config config;
+  void SchedulerTimer()
+  {
+    u32 mail_time_state = 0;
+    u32 download_time_state = 0;
+    Common::SetCurrentThreadName("KD Timer");
+    while (true)
+    {
+      std::unique_lock lg(m_scheduler_lock);
+      if (m_mail_span.load() <= mail_time_state)
+      {
+        m_scheduler_work_queue.EmplaceItem([this] { SchedulerWorker(SchedulerEvent::MAIL); });
+        INFO_LOG_FMT(IOS_WC24, "NET_KD_REQ: Dispatching Mail Task from Scheduler");
+        mail_time_state = 0;
+      }
+
+      if (m_download_span.load() <= download_time_state)
+      {
+        INFO_LOG_FMT(IOS_WC24, "NET_KD_REQ: Dispatching Download Task from Scheduler");
+        m_scheduler_work_queue.EmplaceItem([this] { SchedulerWorker(SchedulerEvent::DOWNLOAD); });
+        download_time_state = 0;
+      }
+
+      m_scheduler_cond_var.wait_for(lg, std::chrono::minutes{1}, [&] { return m_shutdown; });
+      lg.unlock();
+
+      if (m_shutdown)
+        return;
+
+      mail_time_state++;
+      download_time_state++;
+    }
+  }
+
+  enum class SchedulerEvent
+  {
+    MAIL,
```

Comment:
```suggestion
    Mail,
```

Enum class values do not use SCREAMING_CASE